mh/server.py

```diff
@@ -266,13 +277,20 @@ def _worker_target(self, work_queue):
 
             if handler.is_finished():
                 continue
-
-            assert event == selectors.EVENT_READ
-
-            try:
-                handler.on_packet(packet)
-            except Exception as e:
-                handler.on_exception(e)
+            
+            if event == selectors.EVENT_WRITE:
+                try:
+                    if handler.session.layer > 0:  # in-game only
+                        handler.send_packet(packet[0], packet[1], packet[2])
```

Comment:
If packet is a tuple of 3 elements, use the `*` operator instead: `*packet`.

Reply:
Done